zetta_utils/mazepa_addons/configurations/execute_on_gcp_with_sqs.py

```diff
@@ -52,106 +55,124 @@ def _ensure_required_env_vars():
             job_spec=job_spec,
             secrets=secrets,
-            max_replicas=worker_replicas,
+            max_replicas=group.replicas,
             queue=task_queue,
         )
         ctx_managers.append(scaled_job_ctx_mngr)
     else:
         deployment = k8s.get_mazepa_worker_deployment(
             execution_id,
-            image=worker_image,
+            image=image,
             task_queue_spec=task_queue_spec,
             outcome_queue_spec=outcome_queue_spec,
-            replicas=1 if sqs_based_scaling else worker_replicas,
-            resources=worker_resources,
+            replicas=group.replicas,
+            resources=group.resource_limits,
             env_secret_mapping=env_secret_mapping,
-            labels=worker_labels,
-            resource_requests=worker_resource_requests,
-            num_procs=num_procs,
-            semaphores_spec=semaphores_spec,
-            provisioning_model=provisioning_model,
+            labels=group.labels,
+            resource_requests=group.resource_requests,
+            num_procs=group.num_procs,
+            semaphores_spec=group.semaphores_spec,
+            provisioning_model=group.provisioning_model,
         )
         deployment_ctx_mngr = k8s.deployment_ctx_mngr(
             execution_id,
-            cluster_info=worker_cluster,
+            cluster_info=cluster,
             deployment=deployment,
             secrets=secrets,
         )
         ctx_managers.append(deployment_ctx_mngr)
-    return task_queue, outcome_queue, ctx_managers
+    return task_queue, ctx_managers
+
+
+def get_gcp_with_sqs_config(
+    execution_id: str,
+    image: str,
+    groups: Iterable[WorkerGroup],
+    cluster: k8s.ClusterInfo,
+    ctx_managers: list[AbstractContextManager],
+) -> tuple[PushMessageQueue[Task], PullMessageQueue[OutcomeReport], list[AbstractContextManager]]:
+    task_queues = []
+    secrets, env_secret_mapping = k8s.get_secrets_and_mapping(execution_id, REQUIRED_ENV_VARS)
+
+    outcome_queue_name = f"run-{execution_id}-outcome"
+    outcome_queue_spec = {"@type": "SQSQueue", "name": outcome_queue_name, "pull_wait_sec": 2.5}
+    outcome_queue = builder.build(outcome_queue_spec)
+    ctx_managers.append(aws_sqs.sqs_queue_ctx_mngr(execution_id, outcome_queue))
+
+    for group in groups:
+        task_queue, group_ctx_managers = _get_group_taskqueue_and_contexts(
+            execution_id, image, group, cluster, outcome_queue_spec, secrets, env_secret_mapping
+        )
+        task_queues.append(task_queue)
+        ctx_managers.extend(group_ctx_managers)
+
+    return TaskRouter(task_queues), outcome_queue, ctx_managers
 
 
 @builder.register("mazepa.execute_on_gcp_with_sqs")
 def execute_on_gcp_with_sqs(  # pylint: disable=too-many-locals
     target: Union[mazepa.Flow, mazepa.ExecutionState],
     worker_image: str,
-    worker_replicas: int,
-    worker_labels: Optional[Dict[str, str]] = None,
+    worker_groups: dict[str, WorkerGroup],
```

Comment:
Can we make this take a TypedDict instead of a worker group object? Otherwise the specs will bloat